docs/user_guide.md

```diff
@@ -10,4 +10,51 @@ Cyclone provides a powerful workflow engine that serves general workflow purpose
 $ make run_examples SCENE=cicd REGISTRIES=<registry>/<project>
 ```
 
-In addition to YAML format manifest files, you can also create your workflows via APIs provided by Cyclone Server. Refer to [API Docs](./swagger-api-docs.md) for detailed APIs.
\ No newline at end of file
+In addition to YAML format manifest files, you can also create your workflows via APIs provided by Cyclone Server. Refer to [API Docs](./swagger-api-docs.md) for detailed APIs.
+
+## Integration
+
+Cyclone provides an integration center for users to integrate external system information, including sensitive information such as `username`, `password`, `token`, etc,
+and Cyclone will store them by Kubernetes `Secret`.
+
+Cyclone supports 5 types of integrations: `SCM`, `Cluster`, `DockerRegistry`, `SonarQube` and `General`, and SCM is subdivided into the following subclasses:
+`GitHub`, `GitLab`, `BitBucket` and `SVN`.
+
+Supported external systems including:
+
+|  Name  | Sub Class | Supported Versions | Note  |
+| :---:  | :---:     | :---:              | :---: |
+| SCM    | GitHub    | enterprise edition is unsupported |  |
+|        | GitLab    | \>= 8.13.6         | May [affect 10.6 and later versions](#GitLab-Webhooks) to create a SCM webhook automatically triggered pipeline |
+|        | BitBucket | \>= 5.0, BitBucket cloud is unsupported | Different edition may affect [BitBucket Token and Webhooks](#BitBucket-Token-and-Webhooks) |
+|        | SVN       | All                | |
+| Cluster |          | All                | |
+| DockerRegistry |   | All                | |
+| SonarQube |        | Unsupported by now | |
+| General |          | All                | |
+
+### GitLab Webhooks
+
+> Starting with GitLab **10.6**, all Webhook requests to the current GitLab instance server address and/or in a private network will be forbidden by default.
+That means that all requests made to 127.0.0.1, ::1 and 0.0.0.0, as well as IPv4 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16 and IPv6 site-local (ffc0::/10) addresses won’t be allowed.
+
+This behavior can be overridden by enabling the option “Allow requests to the local network from hooks and services”, accessing address:
+- Version 10.6 - 11.3.x , `{your-gitlab-server}/admin/application_settings`
+- Version 11.4.x and later , `{your-gitlab-server}/admin/application_settings/network`
+
+More detailed information: [Webhooks and insecure internal web services](https://docs.gitlab.com/ee/security/webhooks.html)
+
+### BitBucket Token and Webhooks
+
+#### Tokens
+Personal access tokens are supported starting with **5.5**, so in the previous version, you can just use username/password for auth,
```

Comment:
Personal access tokens are supported from **5.5**, so for previous versions, you can only use username/password for auth.